applications/StructuralMechanicsApplication/custom_elements/shell_thin_element_3D4N.cpp

```diff
@@ -2003,18 +2001,19 @@ namespace Kratos
 			double l_13 = std::sqrt(inner_prod(r_13, r_13));
 
 			Vector e_24 = Vector(r_24 / l_24);
-			Vector vec1 = Vector(MathUtils<double>::CrossProduct(r_13*-1.0, e_24));
-			Vector vec2 = Vector(MathUtils<double>::CrossProduct(r_13, e_24));
-			double d_24 = std::sqrt(inner_prod(vec1, vec2));
-			double d_13 = std::sqrt(inner_prod(vec1, vec2));
-			double chi_24 = d_24 / 2.0 / l_24;
-			double chi_13 = d_13 / 2.0 / l_13;
-
-			double chi_xi_t = l_eta / l_xi;
-			double chi_eta_t = l_xi / l_eta;
-
-			double chi_xi_hat = 0.0;
-			double chi_eta_hat = 0.0;
+			const Vector vec1 = Vector(MathUtils<double>::CrossProduct(r_13*-1.0, e_24)); // NOTE: This operation is marked as deprecated by the compiler, but it is not deprecated...not my fault

```

Comment:
what do you mean? if you get the deprecation warning you need to change this. Also as a side remark, multiplying by -1 is quite inefficient in this context (you could use -r_13)

Reply:
i mean, i would change this as

     Vector vec1(3), vec2(3);
     MathUtils<double>::CrossProduct(vec1,e_24,-r_13 );
     MathUtils<double>::CrossProduct(vec2,e_24,r_13);

however this code is sort of stupid. since vec2 = -vec1...

Reply:
@RiccardoRossi , I already tried that, it doesn't work. This code is not mine, feel free of correct it and check like it doesn't behave as expected, so probably there is some kind of memory error

Reply:
it will work now because of the templating we introduced (an expression will be passed in this case)